### ISSUE TITLE
XAuthority should follow XDG standards by default on XDG-Platorms

### ISSUE THREAD

**Issue by Thaodan** [sddm/sddm#1621]
I think  on platforms where XDG is used, the XAuthirty file should go by default to XDG paths.
In this case it is probably XDG_RUNTIME_DIR.

I'm not sure if this a priority after reading earlier bugs such as #944 thou.

**Comment by Vogtinator** [comment]
Currently x11 root sessions use `/tmp` instead of `$XDG_RUNTIME_DIR` for technical reasons. Wayland and x11-user sessions use `$XDG_RUNTIME_DIR`.

**Comment by 62832** [comment]
> Wayland and x11-user sessions use `$XDG_RUNTIME_DIR`.

Does that mean it's possible to `export XAUTHORITY="$XDG_RUNTIME_DIR"/Xauthority` to move `.Xauthority` away from `$HOME`?

**Comment by Vogtinator** [comment]
SDDM sets that variable itself, it does not consume it.

If you use sddm after the move to libxau, the location is outside of ~ already. If it's older, you can set `[X11] UserAuthFile`.

**Comment by 62832** [comment]
Seems like on my end this still defaults to `$HOME/.Xauthority` if I omit the `UserAuthFile` setting.

If it helps, I'm currently using SDDM 0.19.0 on Artix Linux as part of KDE Plasma.

**Comment by Vogtinator** [comment]
Yes, that's the expected behaviour. Released versions don't use libxau yet.

**Comment by artixnous** [comment]
sddm 0.20 creates the xauthority cookie in `/tmp`, ignoring the value set in `/etc/sddm.conf` (`UserAuthFile=.Xauthority`). Is this the expected behaviour?

**Comment by Vogtinator** [comment]
Yes.

**Comment by artixnous** [comment]
Thanks, so if I understand correctly this is imposed by libxau for root X11 sessions and can't be yet/ever be overriden in `/etc/sddm.conf`?

**Comment by Vogtinator** [comment]
Not by libxau itself, but the implementation of Xauth handling in SDDM. It's possible that in the future it may move into `$XDG_RUNTIME_DIR` again.
